src/wallet/wallet.cpp

```diff
@@ -374,7 +380,11 @@ bool CWallet::AddWatchOnly(const CScript& dest)
     const CKeyMetadata& meta = m_script_metadata[CScriptID(dest)];
     UpdateTimeFirstKey(meta.nCreateTime);
     NotifyWatchonlyChanged(true);
-    return WalletBatch(*database).WriteWatchOnly(dest, meta);
+    if (WalletBatch(*database).WriteWatchOnly(dest, meta)) {
+        UnsetWalletFlag(WALLET_FLAG_BLANK_WALLET);
```

Comment:
In commit "[wallet] Support creating a blank wallet" (1cdb8d8346649434450e410971cc3d68c3504a25)

It looks like if you import a watch-only key into a blank wallet WalletModel::canGetAddresses() will start returning true. I wonder if canGetAddresses should be using IsBlank.

Reply:
I think it should. I've added `isBlank()` to the wallet interface and `canGetAddresses()` calls it instead of checking the wallet flag.